akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala

```diff
@@ -256,6 +256,26 @@ final class EventEnvelope[Event](
     metadata.toJava
   }
 
+  /**
+   * Java API: Drop metadata of the given type if present
+   */
+  def removeMetadata(metadataType: Class[_]): EventEnvelope[Event] =
+    removeMetadata(ClassTag(metadataType))
+
+  /**
+   * Scala API: Drop metadata of the given type if present
+   */
+  def removeMetadata[M](implicit classTag: ClassTag[M]): EventEnvelope[Event] = {
+    internalEventMetadata match {
+      case Some(c: CompositeMetadata) =>
+        val filtered = c.entries.filter(_.getClass == classTag.runtimeClass)
```

Comment:
shouldn't this be `filterNot`, since we are deleting
(and missing test for the CompositeMeta case ?)

Reply:
Indeed, I was sure I did add test coverage for both, something must be off with that.